Java/Graphs/KruskalsAlgorithm.java

```diff
@@ -0,0 +1,153 @@
+        // A utility function to find set of an element i 
+        // (uses path compression technique)
+        public int find(int node){
+            if(parents[node] != node){
+                parents[node] = find(parents[node]);
+            }
+            
+            return parents[node];
+        }
+        
+        // A function that does union of two sets of x and y 
+        // (uses union by rank) 
+        public void union(int x, int y){
+            x = find(x);
+            y = find(y);
+            
+            if(rank[x] < rank[y]){
+                parents[x] = y;
+            }else if(rank[x] > rank[y]){
+                parents[y] = x;
+            }else{
+                parents[x] = y;
+                rank[y]++;
+            }
+        }
+    }
+    
+    // Function to find the MST using Kruskal's Algorithm
+    static void spanningTree(int V, int E, ArrayList<ArrayList<Integer>> graph) {
+        // Creating list of edges
+        List<Edge> edges = new ArrayList<>();
+        
+        for(int i = 0; i < V; i++){
+            for(int j = 0; j < V; j++){
+                if(graph.get(i).get(j) != Integer.MAX_VALUE){
+                    edges.add(new Edge(i, j, graph.get(i).get(j)));
+                }
+            }
+        }
+        
+        // Sorting edges according to weights
+        Collections.sort(edges, (x, y) -> x.weight - y.weight);
+        
+        // Index used to pick next edge
+        int e = 0;
+        UnionFind uf = new UnionFind(V);
+        
+        // Resulted graph
+        Edge result[] = new Edge[V];
+        
+        for(Edge next: edges){
+            // Number of edges to be taken is equal to V-1
+            if(e == V-1) break;
+            
+            int x = uf.find(next.u);
+            int y = uf.find(next.v);
+            
+            // If including this edge does't cause cycle, 
+            // include it in result and increment the index  
+            // of result for next edge
+            if(x != y){
+                uf.union(next.u, next.v);
+                result[e++] = next;
+            }
+        }
+        
+        // print for edge and weight
+        System.out.println("Minimum Spanning Tree formed.");
+        System.out.println("Edge : Weight");
+
+        for(int i = 0; i < e; i++){
+            System.out.println(result[i].toString());
+        }
+    }
+
+    public static void main(String[] args) throws IOException{
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
+        System.out.println("Enter no. of Vertices: ");
+        int verticeNo = Integer.parseInt(br.readLine());
+
+        System.out.println("Enter no. of Edges: ");
+        int edgeNo = Integer.parseInt(br.readLine());
+
+        // Constructing graph in the form of adjacency matrix
+        ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
+        for (int i = 0; i < verticeNo; i++) {
+            ArrayList<Integer> temp = new ArrayList<>();
```

Comment:
Tip : Use of array is more encouraged if you want to choose adjacency matrix representation.